lib/handleRequest.js

```diff
@@ -108,9 +108,17 @@ function preHandlerCallback (err, request, reply) {
     return
   }
 
-  var result = reply.context.handler(request, reply)
-  if (result && typeof result.then === 'function') {
-    wrapThenable(result, reply)
+  try {
+    const result = reply.context.handler(request, reply)
+    if (result) {
```

Comment:
I think here you should check specifically for `undefined`, as returning `null`, `false` or `0` will produce a valid JSON document (or a plain text string).

Reply:
done